iOS/DuckDuckGo/MainViewController.swift

```diff
@@ -3042,24 +3042,24 @@ extension MainViewController: UIDropInteractionDelegate {
 // MARK: - VoiceSearchViewControllerDelegate
 
 extension MainViewController: VoiceSearchViewControllerDelegate {
-    
-    func voiceSearchViewController(_ controller: VoiceSearchViewController, didFinishQuery query: String?, target: VoiceSearchTarget) {
-        controller.dismiss(animated: true, completion: nil)
-        guard let query = query else { return }
 
-        DispatchQueue.main.async { [weak self] in
-            guard let self = self else { return }
+    func voiceSearchViewController(_ controller: VoiceSearchViewController, didFinishQuery query: String?, target: VoiceSearchTarget) {
```

Comment:
@afterxleep I added this change to resolve an issue with a conflict between modals being presented. If VoiceSearch was being dismissed while Duck.AI was being presented, there was occasionally an issue with setting the webview traits. This was likely due to the system retrieving the frontmost modal for the collection trait and becoming confused in the process. (See asana issue [here](https://app.asana.com/0/0/1209536356137659/1209563224570424/f))

This fix essentially respects the dismiss callback and only initiates the follow-up action once the dismissal is complete.